netbox/extras/views.py

```diff
@@ -1157,7 +1157,7 @@ def get_required_permission(self):
         return 'extras.view_script'
 
     def get(self, request, module, name):
-        module = get_object_or_404(ScriptModule.objects.restrict(request.user), file_path__startswith=module)
+        module = get_object_or_404(ScriptModule.objects.restrict(request.user), file_path__startswith=module + ".py")
```

Comment:
I would think iif we are doing ".py" at the end it effectively turns it into `file_path=module + ".py"`

Maybe we should remove `__startswith` if that is what we are going for

Reply:
I think it would make sense to remove the `__startswith` as the name should be exactly right.
Also, in the weird case if a file would be named `something.py.py` and a second one with `something.py` is created, it would lead to the same error as is in #14025 again

Reply:
If you have the script "test.py" module comes in as "test" and file_path is "test.py" which is why the startswith is there.  If you remove startswith it won't match.  I think it would be difficult at this point to change module to keep the full name with .py as I think it is stored that way.  Will check into it more...

Reply:
> If you have the script "test.py" module comes in as "test" and file_path is "test.py" which is why the startswith is there. If you remove startswith it won't match. I think it would be difficult at this point to change module to keep the full name with .py as I think it is stored that way. Will check into it more... Thinking about it, it might make more sense to remove a potential ".py" from file_path if there is one - that should be pretty safe.

Yeah, but we are matching on file_path, so if you are doing `{module}.py` it is either going to match or not with or without `__startswith`.

Not sure if I am explaining this well, if you need more info hit me up.  I still think startrswith is not needed with the proposed code change and it will be looking for partial matches when a partial match is not even possible.